api/src/main/java/io/minio/MinioClient.java

```diff
@@ -4295,20 +4295,53 @@ public void putObject(
    * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
    * @throws XmlParserException thrown to indicate XML parsing error.
    */
+  @Deprecated
   public String getBucketPolicy(String bucketName)
       throws BucketPolicyTooLargeException, ErrorResponseException, IllegalArgumentException,
           InsufficientDataException, InternalException, InvalidBucketNameException,
           InvalidKeyException, InvalidResponseException, IOException, NoSuchAlgorithmException,
           XmlParserException {
+    return getBucketPolicy(GetBucketPolicyArgs.builder().bucket(bucketName).build());
+  }
+
+  /**
+   * Gets bucket policy configuration of a bucket.
+   *
+   * <pre>Example:{@code
+   * String config =
+   *     minioClient.getBucketPolicy(GetBucketPolicyArgs.builder().bucket("my-bucketname").build());
+   * }</pre>
+   *
+   * @param args {@link GetBucketPolicyArgs} object.
+   * @return String - Bucket policy configuration as JSON string.
+   * @throws BucketPolicyTooLargeException thrown to indicate returned bucket policy is too large.
+   * @throws ErrorResponseException thrown to indicate S3 service returned an error response.
+   * @throws IllegalArgumentException throws to indicate invalid argument passed.
+   * @throws InsufficientDataException thrown to indicate not enough data available in InputStream.
+   * @throws InternalException thrown to indicate internal library error.
+   * @throws InvalidBucketNameException thrown to indicate invalid bucket name passed.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public String getBucketPolicy(GetBucketPolicyArgs args)
+      throws BucketPolicyTooLargeException, ErrorResponseException, IllegalArgumentException,
+          InsufficientDataException, InternalException, InvalidBucketNameException,
+          InvalidKeyException, InvalidResponseException, IOException, NoSuchAlgorithmException,
+          XmlParserException {
+    if (args == null) {
+      throw new IllegalArgumentException("null arguments");
+    }
```

Comment:
This can be replaced with `checkArgs(args);` after rebasing with latest master branch

Reply:
Done